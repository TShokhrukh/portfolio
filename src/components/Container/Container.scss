
.Container {
  margin-left: auto;
  margin-right: auto;
}

@each $breakpoint, $max-width in $container-max-widths {
  @include media-breakpoint-up($breakpoint) {
    .Container {
      max-width: #{$max-width};
    }
  }
}
