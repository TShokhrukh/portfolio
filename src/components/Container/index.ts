export * from './build'
export * from './__types__'
export { Container as ContainerBase } from './Container'
