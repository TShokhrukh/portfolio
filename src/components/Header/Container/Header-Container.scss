.Header-Container {
  border-radius: var(--header-container-border-radius);
  background-color: var(--header-container-background);
  display: flex;
  gap: var(--header-container-gap);
  padding: var(--header-container-padding-y) var(--header-container-padding-x);
}
