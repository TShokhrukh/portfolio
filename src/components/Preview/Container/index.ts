export { PreviewContainer as default } from './Preview-Container'
