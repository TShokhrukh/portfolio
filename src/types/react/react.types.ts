import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
