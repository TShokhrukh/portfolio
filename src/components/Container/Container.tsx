import React from 'react'
import { cn } from '@bem-react/classname'
import type { IContainerProps } from './__types__'
import './Container.scss'

export const cnContainer = cn('Container')

export const Container: React.FC<IContainerProps> = ({
  id,
  className,
  children,
  ...props
}) => {
  return (
    <div {...props} id={id} className={cnContainer({}, [className])}>
      {children}
    </div>
  )
}
