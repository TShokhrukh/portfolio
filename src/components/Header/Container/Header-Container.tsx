import { withBemMod } from '@bem-react/core'
import { Container } from '~/components/Container/build'
import { cnHeader } from '../Header.const'

import './Header-Container.scss'

export const HeaderContainer = withBemMod(cnHeader('Container'), {})(Container)
