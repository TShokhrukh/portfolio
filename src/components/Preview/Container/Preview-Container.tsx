import { withBemMod } from '@bem-react/core'
import { Container } from '~/components/Container'
import { cnPreview } from '../Preview.const'
import './Preview-Container.scss'

export const PreviewContainer = withBemMod(cnPreview('Container'), {})(Container)
