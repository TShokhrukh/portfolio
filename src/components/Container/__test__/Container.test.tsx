/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { Container } from '../Container'

describe('Container', () => {
  test('component template', () => {
    render(<Container data-testid='container' />)

    expect(screen.getByTestId('container')).toMatchSnapshot()
  })

  test('component render children', () => {
    render(<Container data-testid='container'>children</Container>)

    expect(screen.getByTestId('container')).toHaveTextContent('children')
  })
})
