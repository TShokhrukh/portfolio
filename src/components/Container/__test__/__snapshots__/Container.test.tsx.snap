// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component template 1`] = `
<div
  class="Container"
  data-testid="container"
/>
`;
